### PR TITLE
LP-2027 Match capital contribution max length to CHIPS

### PR DESCRIPTION
**Pull request by jdowie-ch**
### JIRA link

https://companieshouse.atlassian.net/browse/LP-2027

### Change description

CHIPS enforces CheckMaxDecimalValueRule(precision=10, scale=2), allowing a maximum of 8 digits before the decimal point and 2 after (max value 99999999.99). The web form previously allowed up to 12 digits before the decimal, so values accepted on Web were rejected downstream by CHIPS.

Add a max-digits check to the capital contribution validation that counts significant digits (ignoring leading zeros, matching the CHIPS BigDecimal precision) and rejects anything over 99999999.99 with a new maxValue error message (EN + CY). Add unit tests for the new boundary and valid cases.

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by markpit — APPROVED** [review]

File: locales/cy/errors.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
             "valueRequired": "WELSH - Enter the total value of all monetary and non-monetary contributions",
             "twoDecimalPlaces": "WELSH - Value must be a number with 2 decimal places",
             "moreThanZero": "WELSH - Value must be 0.01 or more",
+            "maxValue": "WELSH - Value must be 99999999.99 or less",
```
Comment (vschot) [inline locales/cy/errors.json]: These validation message's dont match with the message on the API... I think in the ticket it mentions they should match

Comment (jdowie-ch) [reply locales/cy/errors.json]: Updated to match the corresponding message in the API

**Review by vschot — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: locales/en/errors.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
             "valueRequired": "Enter the total value of all monetary and non-monetary contributions",
             "twoDecimalPlaces": "Value must be a number with 2 decimal places",
             "moreThanZero": "Value must be 0.01 or more",
-            "maxValue": "Value must be 99999999.99 or less",
+            "maxValue": "The capital contribution value cannot have more than 8 digits before the decimal",
```
Comment (mwestacott) [inline locales/en/errors.json:7]: A requirements question really, but it sounds (feels like) the word 'point' should be present at the end of this error message.

Comment (jdowie-ch) [reply locales/en/errors.json:7]: I copied the preexisting wording from the current message in the API (`src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/controller/LimitedPartnerControllerUpdateTest.java line 304`) the other related error messages use "decimal number". I think it's a good point about checking what the preferred wording should be 👍

Comment (mwestacott) [reply locales/en/errors.json:7]: Yep. I saw that the CHIPS error message references '2 decimal places'.
